tests/rptest/services/action_injector.py

```diff
@@ -0,0 +1,183 @@
+import dataclasses
+import random
+import time
+from threading import Thread
+from typing import Optional, List
+
+from ducktape.cluster.cluster import ClusterNode
+from ducktape.utils.util import wait_until
+
+from rptest.clients.types import TopicSpec
+from rptest.services.admin import Admin
+from rptest.services.failure_injector import FailureSpec, FailureInjector
+from rptest.services.redpanda import RedpandaService
+
+
+@dataclasses.dataclass
+class ActionConfig:
+    cluster_start_lead_time_sec: float
+    min_time_between_actions_sec: float
+    max_time_between_actions_sec: float
+    max_affected_nodes: Optional[int] = None
+
+    def random_time_in_range(self) -> float:
+        return random.uniform(self.min_time_between_actions_sec, self.max_time_between_actions_sec)
+
+
+class RandomNodeOp:
```

Comment:
Thanks for the suggestion, I see the name is a bit disambiguous. I decided to rename it to `DisruptiveAction` which I think is closer to the intent but suggestions are welcome. I think `PeriodicAction` could be used as well but it does not convey _what_ the action does. 

I did not want to name it something related to Failure because the action could be something which is not failure but simply adding a topic or deleting a topic, or leadership transfer etc.

> If you move node selection to ActionInjectorThread and just pass a node to the action method you'll reduce cognitive load: less methods => less possible interactions.

I had node selection in the thread in an early version of the code, but I moved it to the action because in some cases node may not be required for the action being performed eg creating a topic. So we let the action decide what it wants to do for node selection. 

There are fields in `DisruptiveAction` which are node specific but it is up to the action to use them or not.

Reply:
I've looked through the changes and in all the cases we use the base class's target so I can't find the case you mention.

Reply:
It was the case for leadership transfer [here](https://github.com/redpanda-data/redpanda/commit/ce5caaeb0b749242ccf9b8b3634f64d813b1927e#diff-d0b45de9466a84e1f2fb8c4de043a364b399cc796c2653b84922ad353c49a3feR69) which I removed from this PR but I am working on in another PR.

Node will also not be required for topic create and delete action which I have to work on next. Neither will require node to be passed to action.

Reply:
I still see `RandomNodeOp` in the first commit instead of `DisruptiveAction`; you rename RandomNodeOp to DisruptiveAction only in 4th commit "tests: attempt to restore state on exit". Please edit the history and do not introduce `RandomNodeOp` in the first place. This comment also applies to `__call__`, `limit_reached` and other similar issues